.ci.yaml

```diff
@@ -12,6 +12,17 @@ platform_properties:
   linux:
     properties:
       os: Linux
+  linux_android:
+    properties:
+      os: Ubuntu
+      cores: "8"
+      device_type: none
+      dependencies: >-
+        [
+          {"dependency": "android_sdk", "version": "version:33v6"},
+          {"dependency": "open_jdk", "version": "version:11"},
```

Comment:
These were taken from flutter/flutter's configs. I believe it's similar to what we are currently using in Cirrus, so hopefully it'll transition smoothly.

Reply:
Can we use a different platform name to avoid confusion from framework ones, which needs a tethered android phone? Alternatively, I am wondering if we can consolidate the to-be-added (linux) platforms, and leave the dependencies difference on target level.

Reply:
> Can we use a different platform name to avoid confusion from framework ones, which needs a tethered android phone?

IMO the flutter/flutter naming is confusing, because `linux_android` has one specific device, then there are a bunch of `linux_android_somedevice` variants with different devices. Why does `linux_android` not match the naming scheme there?

Also, I don't expect to need device-specific configs in this repo since we use FTL for device tests, so the naming should be much simpler here.

> Alternatively, I am wondering if we can consolidate the to-be-added (linux) platforms, and leave the dependencies difference on target level.

Do you mean _desktop, _web, and _android? That would mean duplicating all of the generic platform-targeting configuration (These two lines for Android, cmake/clang/etc for desktop, etc.) for every single test, which makes things very redundant and harder to update. I would really like to avoid that.

Unless you mean putting all the things needed for all three into a single Linux configuration. I'm fine with that in terms of file structure, but I assume that slows things down unnecessarily in CI.

Reply:
> Alternatively, I am wondering if we can consolidate the to-be-added (linux) platforms

(And FWIW, I believe this is the last one I will need to add. My plan was one generic one for things like Dart analyze and unit tests that have no target, and one for each of the three targets that Linux hosts are used for.)

Reply:
Different platforms handling different set of tasks are expected, and sound good to me.

We have some metrics data depending on the platform assuming `Linux_android` targets are tied to a phone. Do you mind holding a bit on this PR before I find a good way to handle the metrics?

https://github.com/flutter/packages/pull/4229 LGTM to merge.

Reply:
Created https://github.com/flutter/flutter/issues/129039, I expect to fix it in the coming fixit week with a higher priority.

Reply:
https://github.com/flutter/flutter/issues/129039 is fixed. This should be unblocked.